src/main/kotlin/racingcar/communication/input/InputConsole.kt

```diff
@@ -0,0 +1,35 @@
+package racingcar.communication.input
+
+import java.util.Scanner
+
+class InputConsole : Input {
+    private val scan: Scanner = Scanner(System.`in`)
+
+    override fun numberOfRacingCars(): Int = inputToCorrectIntValue()
+    override fun finalLab(): Int = inputToCorrectIntValue()
+
+    private fun inputToCorrectIntValue(): Int {
+        var inputValue: Int
+
+        do {
+            inputValue = scanIntValue()
+        } while (inputValue == COULD_NOT_READ)
+
+        return inputValue
+    }
+
+    private fun scanIntValue(): Int {
+        kotlin.runCatching {
+            return scan.nextInt()
+        }.onFailure {
+            println(it.message)
+            scan.next()
+        }.getOrElse {
+            return COULD_NOT_READ
+        }
```

Comment:
```suggestion
        }.getOrDefault(COULD_NOT_READ)
```

저번에 getOrThrow 관련해서 말씀 주셨던 것 같은데요
getOrThrow()는 기본적으로 getOrElse { throw it } 으로 정의되어 있습니다.
하지만 만약 exception을 다른 것으로 보내고 싶으실 경우에는 getOrThrow 대신 getOrElse { throw xxxException() } 이렇게 사용하는 것으로 알고 있네요

그리고 onFailure는 Result 타입을 받는 쪽에서 실패에 대한 처리를 할 때 활용합니다.
여기에서는 getOrElse가 적합하지 않고, getOrDefault가 적합해보이네요

Reply:
감사합니다~ 참고할게요!

Reply:
아 `scan.next()` 를 못봤네요 😅